### ISSUE TITLE
@mattwparas hi, i have problem with ur config open-term/xplr

### ISSUE THREAD

**Issue by mattwparas** [mattwparas/helix-config#5]
@mattwparas hi, i have problem with ur config open-term/xplr
<img width="733" alt="image" src="https://github.com/helix-editor/helix/assets/13052752/ed39bd4b-8b15-4d66-967d-b55879467646">
on mac zsh is located /bin/zsh

_Originally posted by @StringNick in https://github.com/helix-editor/helix/issues/8675#issuecomment-2223051565_

**Comment by mattwparas** [comment]
For this, you have to set the default shell, so put this in your init.scm:
```scheme
(require (only-in "term.scm" set-default-shell!))
(set-default-shell! "path/to/my/shell/here")
```

**Comment by StringNick** [comment]
> For this, you have to set the default shell, so put this in your init.scm:
> 
> ```scheme
> (require (only-in "term.scm" set-default-shell!))
> (set-default-shell! "path/to/my/shell/here")
> ```

yes, i put it, temrinal and xplr openning, but after close terminal or try to open file xplr bugging, screen above is for terminal quit

**Comment by mattwparas** [comment]
so just to clarify, I believe the `exit` and the shell path issue are two different things. This should just solve the path to the shell

**Comment by mattwparas** [comment]
The exit behavior is something else from the path issue. I'm attempting to reproduce that but currently unable to

**Comment by mattwparas** [comment]
From a clean slate, what happens if you run :kill-active-terminal instead of exit from the terminal?

**Comment by StringNick** [comment]
> From a clean slate, what happens if you run :kill-active-terminal instead of exit from the terminal?

kill-active-terminal closing terminal and no issues
if i try to open file from xplr, happend just this:
<img width="609" alt="image" src="https://github.com/mattwparas/helix-config/assets/13052752/62afe478-bef4-4d29-8d8f-0e9ccd9d208f">


i just call exit in terminal and helix is crashing after it

**Comment by mattwparas** [comment]
I'll work on the exit issue in the terminal, I'll have to get set up on a mac to try to reproduce, the behavior on linux does not appear to be the same.

For xplr, can you provide a screenshot right before you try to open a file?

**Comment by mattwparas** [comment]
Okay, I was able to reproduce on mac, I'll have a fix later today

**Comment by StringNick** [comment]
@mattwparas 
<img width="1636" alt="image" src="https://github.com/mattwparas/helix-config/assets/13052752/db41d3bf-477b-4214-9f88-0273e2768667">
after
<img width="303" alt="image" src="https://github.com/mattwparas/helix-config/assets/13052752/2fd13296-21cc-490d-81de-1cf4c5fd1af3">

**Comment by StringNick** [comment]
@mattwparas also ctrl - c (interrupt signal not working in terminal)

**Comment by mattwparas** [comment]
I've pushed a possible fix for the xplr issue - let me know if it makes a difference for you since I'm unable to reproduce myself

ctrl-c just isn't implemented yet - I can implement that handler.

It is not possible to open the terminal as a helix window as of now

**Comment by StringNick** [comment]
> I've pushed a possible fix for the xplr issue - let me know if it makes a difference for you since I'm unable to reproduce myself. I've also pushed a fix for ctrl c as well
> 
> It is not possible to open the terminal as a helix window as of now
Same new empty tab with lambda, but now i saw that he is executing some command in terminal (./xplr PATH_TO_FILE)

https://github.com/user-attachments/assets/9ae65a88-8b41-44b6-80d5-46ec9c862a88
